diffrax/_adjoint.py

```diff
@@ -482,7 +484,16 @@ def loop(
         init_state = _nondiff_solver_controller_state(
             self, init_state, passed_solver_state, passed_controller_state
         )
-        inputs = (args, terms, self, kwargs, solver, saveat, init_state)
+        inputs = (
+            args,
+            terms,
+            self,
+            kwargs,
+            solver,
+            saveat,
+            init_state,
+            kwargs.pop("direction"),
```

Comment:
I think it'd be better to just make `direction` be another keyword argument to `loop`.
(Or for that matter could it just be left in `kwargs` and passed through? So no code changes here whatsoever?)

Reply:
Moved to avoid direction as keyword altogether with just modifying the function